<launch>


    <!--arg name="tf_world"  default="0.0 0.0 0.0 0.0 0.0 0.0"/-->
    <arg name="tf_world"  default="40.0 -106.5 -0.32 0.0 0.0 110.5"/>
    <!--arg name="tf_world"  default="39.3 -105.0 -0.32 0.0 0.0 110.5"/-->


    <node name = "rgb_decompress"
          pkg = "image_transport"
          type = "republish"
          args = "compressed in:=/rgb/image_rect
                  raw out:=/rgb/image_rect"
          output = "screen">
    </node>


    <node name = "depth_decompress"
          pkg  = "image_transport"
          type = "republish"
          args = "compressedDepth in:=/depth/image_rect
                  raw out:=/depth/image_rect"
          output = "screen">
    </node>


    <include file = "$(find rtabmap_ros)/launch/rtabmap.launch">

        <arg name = "rgb_topic"               value = "/rgb/image_rect"/>
        <arg name = "depth_topic"             value = "/depth/image_rect"/>
        <arg name = "camera_info_topic"       value = "/rgb/camera_info"/>
        <arg name = "depth_camera_info_topic" value = "/depth/camera_info"/>

        <arg name = "args"                    value = "--delete_db_on_start
                                                       --Optimizer/GravitySigma 0.3
                                                       --Odom/Holonomic true
                                                       --Odom/ResetCountdown 1
                                                       --Rtabmap/StartNewMapOnLoopClosure true
                                                       --Odom/Strategy 1
                                                       --Reg/Force3DoF false
                                                       --Kp/WordsPerImage -1"/>

        <arg name = "queue_size"              value = "10"/>
        <arg name = "approx_sync"             value = "true"/>
        <arg name = "wait_imu_to_init"        value = "false"/>
        <arg name = "visual_odometry"         value = "true"/>

        <arg name = "odom_topic"              value = "/rtabmap/odom"/>
        <arg name = "imu_topic"               value = "/imu/data2"/>
        <arg name = "map_frame_id"            value = "rtab_init"/>
        <arg name = "frame_id"                value = "rtab_pose"/>
        <arg name = "use_sim_time"            value = "true"/>
        <arg name = "rtabmapviz"              value = "true"/>
        <arg name = "rviz"                    value = "false"/>

    </include>


    <node name = "point_cloud_converter"
          pkg = "point_cloud_converter"
          type = "point_cloud_converter_node"
          output = "screen">

          <remap from = "/points2_in" to = "/rtabmap/octomap_occupied_space"/>
          <remap from = "/points_out" to = "/cloud/points"/>

    </node>


    <node name   = "joints_node"
          pkg    = "espeleo_mythings"
          type   = "joints_repub.py"
          output = "screen">
    </node>


    <node name   = "odom_node"
          pkg    = "espeleo_mythings"
          type   = "odom.py"
          args   = ""
          output = "screen">
    </node>


    <node name   = "espeleo_pose_ekf"
          pkg    = "espeleo_pose_ekf"
          type   = "espeleo_pose_ekf"
          args   = ""
          output = "screen">

          <remap from = "/odom"                      to = "/robot_odom"/>
          <remap from = "/imu_data"                  to = "/imu/data3"/>
          <remap from = "/espeleo_pose_ekf/ekf_odom" to = "/ekf/odom"/>

    </node>


    <node name   = "odom_to_init_node"
          pkg    = "espeleo_mythings"
          type   = "odom_to_init.py"
          args   = "_tf_world:='$(arg tf_world)'"
          output = "screen">
    </node>


    <node name   = "odom_to_file_node"
          pkg    = "espeleo_mythings"
          type   = "odom_to_file.py"
          args   = ""
          output = "screen">
    </node>


</launch>
